plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs

```diff
@@ -0,0 +1,120 @@
+import qualified Development.IDE.Core.Shake     as Shake
+import           Development.IDE.GHC.Compat     (HieASTs (HieASTs),
+                                                 RealSrcSpan (..), mkHieFile',
+                                                 mkRealSrcLoc, mkRealSrcSpan,
+                                                 runHsc, srcSpanEndCol,
+                                                 srcSpanEndLine,
+                                                 srcSpanStartCol,
+                                                 srcSpanStartLine, tcg_exports)
+import           Development.IDE.GHC.Error      (realSrcSpanToRange)
+import           GHC.Generics                   (Generic)
+import           HieTypes                       (HieASTs, HieFile)
+import           Ide.Types                      (PluginDescriptor (..),
+                                                 PluginId,
+                                                 defaultPluginDescriptor)
+import qualified Language.LSP.Types             as LSP
+import           Stan.Analysis                  (Analysis (..), runAnalysis)
+import           Stan.Core.Id                   (Id (..))
+import           Stan.Inspection                (Inspection (..))
+import           Stan.Inspection.All            (inspectionsIds, inspectionsMap)
+import           Stan.Observation               (Observation (..))
+
+descriptor :: Recorder (WithPriority Log) -> PluginId -> PluginDescriptor IdeState
+descriptor recorder plId = (defaultPluginDescriptor plId) {pluginRules = rules recorder}
+
+newtype Log = LogShake Shake.Log deriving (Show)
+
+instance Pretty Log where
+  pretty = \case
+    LogShake log -> pretty log
+
+data GetStanDiagnostics = GetStanDiagnostics
+  deriving (Eq, Show, Generic)
+
+instance Hashable GetStanDiagnostics
+
+instance NFData GetStanDiagnostics
+
+type instance RuleResult GetStanDiagnostics = ()
+
+rules :: Recorder (WithPriority Log) -> Rules ()
+rules recorder = do
+  define (cmapWithPrio LogShake recorder) $
+    \GetStanDiagnostics file -> do
+      hie <- fromJust <$> getHieFile file
+      let enabledInspections = HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)]
+      let analysis = runAnalysis Map.empty enabledInspections [] [hie]
+      return (analysisToDiagnostics file analysis, Just ())
+
+  action $ do
+    files <- getFilesOfInterestUntracked
+    void $ uses GetStanDiagnostics $ HM.keys files
+  where
+    analysisToDiagnostics :: NormalizedFilePath -> Analysis -> [FileDiagnostic]
+    analysisToDiagnostics file = mapMaybe (observationToDianostic file) . toList . analysisObservations
+    observationToDianostic :: NormalizedFilePath -> Observation -> Maybe FileDiagnostic
+    observationToDianostic file (Observation {observationSrcSpan, observationInspectionId}) =
+      do
+        inspection <- HM.lookup observationInspectionId inspectionsMap
+        let
+          -- Looking similar to Stan CLI output
```

Comment:
Is there a reason to avoid `prettyShowInspection` of `Stan.Inspection`? If so, could you expand this comment with the reason?

Reply:
`prettyShowInspection` included "Id" (which is already in _code) and "Severity" (which could be placed in _severety), which are not needed to show here, as I think.

Reply:
Worth a comment.

Reply:
Comment done.